### ISSUE TITLE
A hang at compiling the yt-dlp binary on macOS 10.14 "Mojave"

### ISSUE THREAD

**Issue by scrutinizer11** [yt-dlp/yt-dlp#3915]
### Checklist

- [X] I'm asking a question and **not** reporting a bug/feature request
- [X] I've looked through the [README](https://github.com/yt-dlp/yt-dlp#readme)
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar questions including closed ones

### Question

Hello,

I installed **Python 3.10.4**. The default installation directory is _/Library/Frameworks/Python.framework/Versions/3.10/bin/python3_. However, the compilation stalled at the line 
> (reading log messages from standard input)" preceded by "error: unable to read SDK settings for '/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk'. 

I run Mojave, so I assume the version installed on my Mac is **MacOSX10.14.sdk**. Can I have a workaround or a compatible version of the source code?

The log until the hang is:

```bash
Macbookpro:yt-dlp-master me$ make
COLUMNS=80 /usr/bin/env python3 yt_dlp/__main__.py --ignore-config --help | /usr/bin/env python3 devscripts/make_readme.py
/usr/bin/env python3 devscripts/make_contributing.py README.md CONTRIBUTING.md
/usr/bin/env python3 devscripts/make_issue_template.py .github/ISSUE_TEMPLATE_tmpl/1_broken_site.yml .github/ISSUE_TEMPLATE/1_broken_site.yml
/usr/bin/env python3 devscripts/make_issue_template.py .github/ISSUE_TEMPLATE_tmpl/2_site_support_request.yml .github/ISSUE_TEMPLATE/2_site_support_request.yml
/usr/bin/env python3 devscripts/make_issue_template.py .github/ISSUE_TEMPLATE_tmpl/3_site_feature_request.yml .github/ISSUE_TEMPLATE/3_site_feature_request.yml
/usr/bin/env python3 devscripts/make_issue_template.py .github/ISSUE_TEMPLATE_tmpl/4_bug_report.yml .github/ISSUE_TEMPLATE/4_bug_report.yml
/usr/bin/env python3 devscripts/make_issue_template.py .github/ISSUE_TEMPLATE_tmpl/5_feature_request.yml .github/ISSUE_TEMPLATE/5_feature_request.yml
/usr/bin/env python3 devscripts/make_issue_template.py .github/ISSUE_TEMPLATE_tmpl/6_question.yml .github/ISSUE_TEMPLATE/6_question.yml
/usr/bin/env python3 devscripts/make_supportedsites.py supportedsites.md
git shortlog -s -e -n | awk '!(out[$NF]++) { $1="";sub(/^[ \t]+/,""); print}' > .mailmap
git: error: unable to read SDK settings for '/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk'
git: error: unable to read SDK settings for '/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk'
(reading log message from standard input) 
```

**Comment by pukkandan** [comment]
Not a yt-dlp issue

https://stackoverflow.com/questions/52665394/git-error-unable-to-read-sdk-settings-for-macosx-sdk

**Comment by scrutinizer11** [comment]
I followed that link but the next question arises: if it is not a yt-dlp issue, then how git (that is a binary shipped with Xcode) is aware the 10.15sdk is the requirement in this case? I've been able to compile any binary without the compatibility warning. Why did it stumble over it now?
